doc/specs/#1564 - Settings UI/spec.md

```diff
@@ -0,0 +1,123 @@
+---
+author: Kayla Cinnamon @cinnamon-msft
+created on: 2020-06-29
+last updated: 2020-07-08
+issue id: #1564
+---
+
+# Settings UI Implementation
+
+## Abstract
+
+This spec describes the basic functionality of the settings UI, including disabling it, the navigation items, launch methods, and editing of settings. The specific layout of each page will defined in later design reviews.
+
+## Inspiration
+
+We have been wanting a settings UI since the dawn of Terminal time, so we need to define how it will interact with the application and how users should expect to interact with it.
+
+## Solution Design
+
+The settings UI will be the default experience. We will provide users an option to skip the settings UI and edit the raw JSON file.
+
+### Ability to disable displaying the settings UI
+
+Some users don't want a UI for the settings. We can update the `openSettings` key binding with a `settingsUI` option.
+
+If people still like the UI but want to access the JSON file, we can provide an "Open the JSON file" button at the bottom of the navigation menu.
+
+### Launch methods
+
+#### Proposal 1 - Launch in a new window
+
+Clicking the settings button in the dropdown menu will open the settings UI in a new window. This allows the user to edit their settings and see the Terminal live update with their changes.
+
+In the Windows taskbar, the icon will appear as if Terminal has multiple windows open.
+
+#### Proposal 2 - Launch in a new tab
+
+Clicking the settings button in the dropdown menu will open the settings UI in a new tab. This helps us take steps toward supporting non-terminal content in a tab. However, tab tear-off should be implemented before we have this. This way, users can still see their changes take effect if they want by tearing out the settings UI.
+
+### Editing and saving settings
+
+#### Proposal 1 - Automatically save settings
+
+As users edit fields in the settings UI, they are automatically saved and written to the JSON file. This allows the user to see their settings changes taking place in real time.

```

Comment:
I'm not sure how I feel about this. I typically prefer a conscious "save" button. Think of something like the Settings app on Windows - it doesn't really provide any feedback when you change a setting, even though it auto-saves changes. As a user, I often worry that the settings app didn't actually change the values, because there's no feedback that the value I changed was persisted.

That being said, auto-saving is convenient. I think there just needs to be some sort of _subtle_visual cue that the settings were saved when the user makes changes.

Reply:
I'm just going to aside that it's generally bad to get too far away from the OS default behaviors, so if Windows 10 is auto-saving settings, apps designed for Windows 10 should follow suit or convince MS to go back to the Apply/OK paradigm.

Reply:
My opinion: I liked the Apply/Cancel/Okay button since you can easily revert back if you don't like something.

Reply:
I think there should be an auto-save option. However, there should also be  manual option such as Chips has mentioned. So basically how you save the settings should be an option in settings. That can go under a General or Maintenance Page at the very bottom that's used for directly configuring how you want the Settings UI to look and feel as well how it does things like saving.